networklibrary/src/main/java/com/eis/smsnetwork/broadcast/BroadcastReceiver.java

```diff
@@ -0,0 +1,182 @@
+     * messages. Messages are composed of different fields, separated by
+     * {@link BroadcastReceiver#FIELD_SEPARATOR}. {@link BroadcastReceiver#FIELD_SEPARATOR}s
+     * preceded by a backslash do not separate fields.
+     * Field 0 contains the {@link RequestType} of the request contained in this message.
+     * The rest of the message varies depending on the {@link RequestType}:
+     * <ul>
+     * <li>Invite: there are no other fields</li>
+     * <li>AcceptInvitation: there are no other fields</li>
+     * <li>AddPeer: fields from 1 to the last one contain the phone numbers of each
+     * {@link com.eis.smslibrary.SMSPeer} we have to add to our network</li>
+     * <li>QuitNetwork: there are no other fields, because this request can only be sent by the
+     * {@link com.eis.smslibrary.SMSPeer} who wants to be removed</li>
+     * <li>AddResource: starting from 1, fields with odd numbers contain keys, their following
+     * (even)
+     * field contains the corresponding value</li>
+     * <li>RemoveResource: fields from 1 to the last one contain the keys to remove</li>
+     * </ul>
+     *
+     * @param message The message passed by {@link com.eis.smslibrary.SMSReceivedBroadcastReceiver}.
+     */
+    @Override
+    public void onMessageReceived(SMSMessage message) {
+        Log.d("BR_RECEIVER", "Message received: " + message.getPeer() + " " + message.getData());
+        String[] fields = message.getData().split(SEPARATOR_REGEX);
+        RequestType request;
+        try {
+            request = RequestType.get(fields[0]);
+        } catch (ArrayIndexOutOfBoundsException | NullPointerException e) {
+            return;
+        }
+        if (request == null) return;
+        SMSPeer sender = message.getPeer();
+        SMSNetSubscriberList subscribers =
+                (SMSNetSubscriberList) SMSJoinableNetManager.getInstance().getNetSubscriberList();
+        SMSNetDictionary dictionary =
+                (SMSNetDictionary) SMSJoinableNetManager.getInstance().getNetDictionary();
+        boolean senderIsNotSubscriber = !subscribers.getSubscribers().contains(sender);
+
+        switch (request) {
+            case Invite: {
+                if (fields.length > 1) return;
+                SMSJoinableNetManager.getInstance().checkInvitation(new SMSInvitation(sender));
+                break;
+            }
+            case AcceptInvitation: {
+                if (fields.length > 1) return;
+                // Verifying if the sender has been invited to join the network
+                Set<SMSPeer> invitedPeers = SMSJoinableNetManager.getInstance().getInvitedPeers();
+                if (!invitedPeers.contains(sender))
+                    return;
+                invitedPeers.remove(sender);
+
+                // Sending to the invited peer my subscribers list
+                StringBuilder mySubscribers = new StringBuilder(RequestType.AddPeer.asString() +
```

Comment:
Although is not part of this review, I would have overridden `toString()` method instead of using `asString()` method, supposing that the purpose of this method is the same.

Reply:
`toString()` would return the `String` "AddPeer", and I think that's the default behavior of this method in `enums`, so I guess the method `asString()` was created because somebody might want to call `toString()` and have it behave as it does by default.
Anyway I didn't write class `RequestType`.

![Screenshot from 2020-03-03 16-39-36](https://user-images.githubusercontent.com/56405609/75792033-b3591680-5d6d-11ea-9d3c-73d8f01278a4.png)

Reply:
Thank you for the explanation, it wasn't very clear to me! Nothing to say then.